src/udiskslinuxpartitiontable.c

```diff
@@ -640,32 +303,51 @@ udisks_linux_partition_table_handle_create_partition (UDisksPartitionTable   *ta
       goto out;
     }
 
-  if (!udisks_daemon_launch_spawned_job_sync (daemon,
-                                              object,
-                                              "partition-create", caller_uid,
-                                              NULL, /* GCancellable */
-                                              0,    /* uid_t run_as_uid */
-                                              0,    /* uid_t run_as_euid */
-                                              NULL, /* gint *out_status */
-                                              &error_message,
-                                              NULL,  /* input_string */
-                                              "%s",
-                                              command_line))
+  /* Ensure we _start_ at MiB granularity since that ensures optimal IO...
+   * Also round up size to nearest multiple of 512
+   */
+  start_mib = offset / MIB_SIZE + 1L;
+  end_bytes = start_mib * MIB_SIZE + ((size + 511L) & (~511L));
+  size = end_bytes - (start_mib * MIB_SIZE); /* recalculate size with the new end_bytes */
```

Comment:
None of these is needed with ``BD_PART_ALIGN_OPTIMAL`` which does real optimal alignment not just magic that's supposed to work for every single disk.